src/megaapi_impl.cpp

```diff
@@ -11578,6 +11587,65 @@ void MegaApiImpl::chatlinkjoin_result(error e)
     fireOnRequestFinish(request, megaError);
 }
 
+void MegaApiImpl::folderlinkinfo_result(error e, handle owner, handle ph, string *attr, string* key, m_off_t currentSize, uint32_t numFiles, uint32_t numFolders, m_off_t versionsSize, uint32_t numVersions)
+{
+    MegaError megaError(e);
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_PUBLIC_LINK_INFORMATION)) return;
+
+    if (e == API_OK)
+    {
+        handle h;
+        size_t t = 0;
+        string nodekey;
+        string attrs;
+        byte* buf = NULL;
+        const char* k = NULL;
+        unsigned int keylength = FOLDERNODEKEYLENGTH;
+        byte binkey[FILENODEKEYLENGTH];
+        SymmCipher cipher;
+        SymmCipher* sc = &client->key;
```

Comment:
This is the master key of the logged in account. It cannot be used to decrypt the received key. Instead, you need to use the key part of the folder link. Note you're calling `folderaccess(link)`, which sets the `client->key` to the folder link, but it will break any de/ciphering requiring the account's masterkey.
A possible workaround is to modify `folderaccess()` to receive a `SymmCipher` as parameter and store the folder key in there, rather than in `Client::key`.